Extensions/BitmapText/JsExtension.js

```diff
@@ -0,0 +1,631 @@
+          objectContent[propertyName] = parseFloat(newValue);
+        else objectContent[propertyName] = newValue;
+        return true;
+      }
+
+      return false;
+    };
+    // $FlowExpectedError
+    objectBitmapText.getProperties = function(objectContent) {
+      const objectProperties = new gd.MapStringPropertyDescriptor();
+
+      objectProperties
+        .getOrCreate('text')
+        .setValue(objectContent.text)
+        .setType('textarea')
+        .setLabel(_('Text'));
+
+      objectProperties
+        .getOrCreate('color')
+        .setValue(objectContent.color)
+        .setType('color')
+        .setLabel(_('Base color'));
+
+      objectProperties
+        .getOrCreate('opacity')
+        .setValue(objectContent.opacity.toString())
+        .setType('number')
+        .setLabel(_('Opacity (0-255)'));
+
+      objectProperties
+        .getOrCreate('fontSize')
+        .setValue(objectContent.fontSize.toString())
+        .setType('number')
+        .setLabel(_('Base size'));
+
+      objectProperties
+        .getOrCreate('align')
+        .setValue(objectContent.align)
+        .setType('choice')
+        .addExtraInfo('left')
+        .addExtraInfo('center')
+        .addExtraInfo('right')
+        .setLabel(_('Base alignment'));
+
+      objectProperties
+        .getOrCreate('fontFamily')
+        .setValue(objectContent.fontFamily)
+        .setType('resource')
+        .addExtraInfo('font')
+        .setLabel(_('Base font family'));
+
+      objectProperties
+        .getOrCreate('wordWrap')
+        .setValue(objectContent.wordWrap ? 'true' : 'false')
+        .setType('boolean')
+        .setLabel(_('Word wrapping'));
+
+      objectProperties
+        .getOrCreate('visible')
+        .setValue(objectContent.visible ? 'true' : 'false')
+        .setType('boolean')
+        .setLabel(_('Visible on start'));
+
+      return objectProperties;
+    };
+    objectBitmapText.setRawJSONContent(
+      JSON.stringify({
+        text: 'Bitmap Text test',
+        opacity: 255,
+        fontSize: 20,
+        visible: true,
+        color: '#000000',
+        fontFamily: 'Arial',
+        align: 'left',
+        wordWrap: true,
+      })
+    );
+
+    // $FlowExpectedError
+    objectBitmapText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    // $FlowExpectedError
+    objectBitmapText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BitmapText',
+        _('BitmapText'),
+        _(
+          'Displays a rich text label using Bitmap markup (allowing to set parts of the text as bold, italic, use different colors and shadows).'
```

Comment:
Bitmap is not a markup. Maybe something like `Displays rich text as a sprite by converting a standard font to a Bitmap font`?